### PR TITLE
Add exception handling for download data responses failure

### PR DESCRIPTION
**Pull request by asadmanzoor93**
**Description:**  This PR adds exception handling for download data responses failure

**JIRA:** 
https://edlyio.atlassian.net/browse/EDLY-3692

**Merge checklist:**

- [ ] All reviewers approved
- [ ] Commits are (reasonably) squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

**Note:** Please add screenshots for any viewable change.

### CODE REVIEW

File: lms/djangoapps/courseware/user_state_client.py [hunk 1]
```diff
@@ -425,7 +426,10 @@ def iter_all_for_block(self, block_key, scope=Scope.user_state):
                 if state == {}:
                     continue
 
-                yield XBlockUserState(sm.student.username, sm.module_state_key, state, sm.modified, scope)
+                try:
+                    yield XBlockUserState(sm.student.username, sm.module_state_key, state, sm.modified, scope)
+                except ObjectDoesNotExist:
+                    log.warning("User Not Available: {0}".format(sm.student_id))
```
Comment (zubair-arbi) [inline lms/djangoapps/courseware/user_state_client.py:432]:
```python
log.warning('The student with id %s from student module does not exist', sm.student_id)
```

Also add a new block for try/catch on `sm.student`

**Review by zubair-arbi — CHANGES_REQUESTED** [review]

File: lms/djangoapps/courseware/user_state_client.py (hunk 1 above)
Comment (asadmanzoor93) [reply lms/djangoapps/courseware/user_state_client.py]: Done

File: lms/djangoapps/courseware/user_state_client.py [hunk 2]
```diff
@@ -425,7 +426,14 @@ def iter_all_for_block(self, block_key, scope=Scope.user_state):
                 if state == {}:
                     continue
 
-                yield XBlockUserState(sm.student.username, sm.module_state_key, state, sm.modified, scope)
+                try:
+                    student_username = sm.student.username
+                except ObjectDoesNotExist:
+                    log.warning('The related user in student module record with the student_id %s does not exist',
```
[the remark this answers is not in the document]
Comment (asadmanzoor93) [reply lms/djangoapps/courseware/user_state_client.py]: Done

**Review by rehan99000 — APPROVED** [review]

File: lms/djangoapps/courseware/user_state_client.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 from django.conf import settings
 from django.contrib.auth.models import User
 from django.core.paginator import Paginator
+from django.core.exceptions import ObjectDoesNotExist
```
Comment (burhandodhy) [inline lms/djangoapps/courseware/user_state_client.py:16]: This line should be before `from django.core.paginator import Paginator`

**Review by burhandodhy — APPROVED** [review]
One minor suggestion

**Pull request merged**